lib/house.rb

```diff
@@ -0,0 +1,39 @@
+class House
+  attr_reader :price,
+              :address,
+              :rooms
+
+  def initialize(price, address)
+    @price = price.gsub("$", '').to_i
+    @address = address
+    @rooms = []
+  end
+
+  def add_room (room)
+    @rooms << room
+  end
+
+  def above_market_average?
+  price >= 500000
+  end
+
+  def rooms_from_category(room_type)
+    @rooms.select do |room|
+        room.category == room_type
+  end
+  end
+
+  def area
+    rooms.each do |room|
```

Comment:
this is really close you would first need to set `house_area = 0` above this each statement. Then return it at the end of the method to return the total area